app/controllers/FileChecksController.scala

```diff
@@ -37,39 +43,70 @@ class FileChecksController @Inject() (
     consignmentService
       .fileCheckProgress(consignmentId, request.token.bearerAccessToken)
       .map(_.asJson.noSpaces)
       .map(Ok(_))
   }
 
-  def fileChecksPage(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+  def fileChecksPage(consignmentId: UUID, uploadFailed: Option[String]): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    val token = request.token.bearerAccessToken
     for {
-      fileChecks <- getFileChecksProgress(request, consignmentId)
-      reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
-    } yield {
-      if (fileChecks.isComplete) {
-        Ok(
-          views.html.fileChecksProgressAlreadyConfirmed(
-            consignmentId,
-            reference,
-            applicationConfig.frontEndInfo,
-            request.token.name,
-            isJudgmentUser = false
-          )
-        ).uncache()
-      } else {
-        Ok(views.html.standard.fileChecksProgress(consignmentId, reference, applicationConfig.frontEndInfo, request.token.name))
-          .uncache()
+      reference <- consignmentService.getConsignmentRef(consignmentId, token)
+      result <- uploadFailed match {
+        case Some(TRUE) => handleFailedUpload(consignmentId, reference, isJudgmentUser = false, token)
+        case _          => handleSuccessfulUpload(consignmentId, reference, isJudgmentUser = false)
       }
-    }
+    } yield result
   }
 
-  def judgmentFileChecksPage(consignmentId: UUID): Action[AnyContent] = judgmentTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+  def judgmentFileChecksPage(consignmentId: UUID, uploadFailed: Option[String]): Action[AnyContent] = judgmentTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    val token = request.token.bearerAccessToken
     for {
-      fileChecks <- getFileChecksProgress(request, consignmentId)
-      reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+      reference <- consignmentService.getConsignmentRef(consignmentId, token)
+      result <- uploadFailed match {
+        case Some(TRUE) => handleFailedUpload(consignmentId, reference, isJudgmentUser = true, token)
+        case _          => handleSuccessfulUpload(consignmentId, reference, isJudgmentUser = true)
+      }
+    } yield result
+  }
+
+  private def handleSuccessfulUpload(consignmentId: UUID, reference: String, isJudgmentUser: Boolean)(implicit request: Request[AnyContent]) = {
+    val token = request.token.bearerAccessToken
+    (for {
+      statuses <- consignmentStatusService.getConsignmentStatuses(consignmentId, token)
+      uploadStatus = statuses.find(_.statusType == UploadType.id)
+      alreadyTriggered = uploadStatus.isDefined && uploadStatus.get.value == CompletedValue.value || uploadStatus.get.value == CompletedWithIssuesValue.value
```

Comment:
If the status is `CompletedWithIssues` does the user still see an error message on the page?

Reply:
Ah no it doesn't. I've made some changes to catch this.